### PR TITLE
discord.py 2.0 update (3d914e08->2.0.1)

### PR DESCRIPTION
**Pull request by Jackenmen**
### Description of the changes

The update is long due...

d.py diff: https://github.com/Rapptz/discord.py/compare/3d914e08...v2.0.1

Things to do:

- overall double-checking of changes from d.py diff + cross-check with Danny's weekly changelogs
- consider adding support for command argument descriptions? I think this could probably be done in a separate PR

Open questions:

- Is the proposed approach for handling voice mutes (`[p]voicemute`) and channel mutes in VC (`[p]channelmute`) alright?
  - in this PR, `[p]voicemute` removes only the Speak perm while `[p]channelmute` removes both message sending perms and speak perms

Things that are handled separately:
- hybrid commands business
    - this is done because I think that there's a lot related to it that we have to work on and it doesn't actually make any sense to work on it all in this PR

### Have the changes in this PR been tested?

No

### CODE REVIEW

File: redbot/core/commands/__init__.py [hunk 1]
```diff
@@ -189,4 +189,13 @@
     bot_has_any_role as bot_has_any_role,
     before_invoke as before_invoke,
     after_invoke as after_invoke,
+    CurrentChannel as CurrentChannel,
+    Author as Author,
+    param as param,
+    MissingRequiredAttachment,
```
Comment (Zephyrkul) [inline redbot/core/commands/__init__.py]:
```suggestion
    MissingRequiredAttachment as MissingRequiredAttachment,
```

**Review by TrustyJAID — DISMISSED** [review]
I've been testing and using this update for several months on one bot and over 2 weeks on 2 other bots without any issues so far.

**Comment by Jackenmen** [comment]
Have you looked through the new commits (all of the ones in Oct 1, 2022 section)? Especially this one:
[`433b3ab` (#5709)](https://github.com/Cog-Creators/Red-DiscordBot/pull/5709/commits/433b3ab019b5cf524c5c02e0c092a2029d77d276)

**Comment by TrustyJAID** [comment]
I did look at it and it seems fine. It will require some updates on #5604 since that uses dataclasses instead of a dict but I'll deal with that after this is merged.

**Review by Kowlin — DISMISSED** [review]
Looks good in my eyes after reading every commit.

**Review by leetfin — COMMENTED** [review]
I've been testing with this PR for months now, no issues noted.

**Comment by Drapersniper** [comment]
For the folks planning to say they have been testing this for months, the issue at hand is there have been several recent changes - so any testing prior to these (Oct 1, 2022) is not fully relevant

**Review by Kowlin — APPROVED** [review]
Tested and ran the latest changes, nothing wrong there. You're amazing Jack.

**Pull request merged**